torch/distributed/_composable/checkpoint_activation.py

```diff
@@ -212,6 +212,7 @@ def checkpoint(module: nn.Module, *, use_reentrant: bool = True) -> nn.Module:
         >>> model(torch.zeros(2, 10)).sum().backward()
 
     """
+    torch._C._log_api_usage_once("torch.distributed.checkpoint")
```

Comment:
nit: maybe checkpoint_activation so not to confuse it with state_dict checkpoint.

Reply:
@rohan-varma I personally like checkpoint_activation (or personally I've heard activation_checkpoint more frequently)
but I'm trying to match the API itself (which is called "checkpoint")

Should we rename the API itself? I think the log should match the API

Reply:
cc @mrshenli  WDYT about renaming the checkpoint() API to 'activation_checkpoint' or 'checkpoint_activation'?

Reply:
Yeah, possibly we should rename the API itself as well.